hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -111,6 +111,8 @@ public void doZeroSum(final List<BalanceChange> changes) {
         var validity = OK;
         var autoCreationFee = 0L;
 
+        final var payer = txnCtx.activePayer();
+        var payerBalanceAfterAdjustments = (long) accountsLedger.get(payer, BALANCE);
```

Comment:
Very clean.  But `TransactionalLedger#get()` is, unfortunately, quite expensive---and this would add another `get()` to **every** `CryptoTransfer`, no matter if it does an auto-creation.

I'd suggest initializing,
```
var payerBalanceAfterAdjustments = Long.MIN_VALUE;
```
and then later,
```
        if (validity == OK && autoCreationFee > 0) {
           payerBalanceAfterAdjustments = (payerBalanceAfterAdjustments != Long.MIN_VALUE)
               ? payerBalanceAfterAdjustments
               : (long) accountsLedger.get(txnCtx.activePayer(), BALANCE);
           if (autoCreationFee > payerBalanceAfterAdjustments) {
               validity = INSUFFICIENT_PAYER_BALANCE;
           }
        }
```

Reply:
Much better... changed now as suggested